manim/mobject/geometry.py

```diff
@@ -177,7 +178,10 @@ def position_tip(self, tip, at_start=False):
         else:
             handle = self.get_last_handle()
             anchor = self.get_end()
-        tip.rotate(angle_of_vector(handle - anchor) - PI - tip.tip_angle)
+        angles = cartesian_to_spherical(handle - anchor)
```

Comment:
It'd help with clarity to add a comment explaining the mathematical reasoning here.

Reply:
Is something like this okay?
```py
tip.rotate(angles[2] - PI - tip.tip_angle) # Rotates the tip along the azimuthal
axis = [np.sin(angles[2]), -np.cos(angles[2]), 0] # Obtains the perpendicular of the tip
tip.rotate(-angles[1] + PI / 2, axis=axis) # Rotates the tip along the vertical wrt the axis
```

Reply:
Also, in my pr #1798 introducing the Plane Mobject, i added the docs for the cartesian_to_spherical function. You might want to take a look at that too.

Reply:
Yeah that seems solid 👍.

> Also, in my pr #1798 introducing the Plane Mobject, i added the docs for the cartesian_to_spherical function

Ah I must've missed that 😅 I'll take another look at that PR then.